### DISCUSSION TITLE
How to let nuxi use self-signed certificate?

### DISCUSSION THREAD

**Discussion by mia106dev** [nuxt/nuxt#16096] [Questions]
Hi.
At first, sorry for my poor English :(

When I try to start-up Nuxt3 project by using command `npx nuxt init <project-name>`, cli returns me this Error message.

```
ERROR  Failed to download template from registry: request to https://raw.githubusercontent.com/nuxt/starter/templates/templates/v3.json failed, reason: self signed certificate in certificate chain

at /C:/Users/username/node_modules/nuxi/dist/chunks/init.mjs:15133:11
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at async downloadTemplate (/C:/Users/username/node_modules/nuxi/dist/chunks/init.mjs:15132:20)
  at async Object.invoke (/C:/Users/username/node_modules/nuxi/dist/chunks/init.mjs:15200:15)
  at async _main (/C:/Users/username/node_modules/nuxi/dist/cli.mjs:50:20)
```

Exactly, I'm (We're) using self-signed certificate, cuz we're introducing external firewall system.
So I need to let nuxi approve and use self-signed certificate.
NPM and git are already using self-signed certificate, so they don't have any ploblems (maybe).

```
 $ git config --global --get http.sslcainfo
C:\Users\UserName\Documents\certs\self-signed-cacert.bundle.pem

 $ npm config -g list      
; "global" config from C:\Program Files\nodejs\etc\npmrc

cafile = "C:\Users\UserName\Documents\certs\self-signed-cacert.bundle.pem"
strict-ssl = true
```

If you have any informations, please leave a comment.

thank you.

**Comment by hanool** [comment]
I had same problems and adding new environment variable fixed it.
`NODE_EXTRA_CA_CERTS`= `path\to\your\cert`

Just set same value of npm config cafile,
in this case it's  `C:\Users\UserName\Documents\certs\self-signed-cacert.bundle.pem`
